client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -107,6 +99,38 @@ export const PatternLibrary = ( {
 		},
 	} );
 
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+
+	const handleViewChange = useCallback(
+		( view: 'grid' | 'list' ) => {
+			const searchParams = new URLSearchParams( location.search );
+			const currentView = searchParams.get( 'grid' ) === '1' ? 'grid' : 'list';
```

Comment:
```suggestion
			const currentView = isGridView ? 'grid' : 'list';
```
`isGridView` already holds this info.

Reply:
I thought about that as well, but decided to go this way so as not to introduce more dependencies to the callback, since I was using `useCallback()`.

Reply:
Since I'm dropping the `useCallback()` (see https://github.com/Automattic/wp-calypso/pull/88728#discussion_r1533446841  ), maybe that doesn't hold anymore.